### ISSUE TITLE
javascript API: client.msearch document reference is incorrect

### ISSUE THREAD

**Issue by boysmovie** [elastic/elasticsearch-js#253]
https://www.elastic.co/guide/en/elasticsearch/client/javascript-api/current/api-reference.html
https://www.elastic.co/guide/en/elasticsearch/client/javascript-api/current/api-reference-1-4.html#api-msearch-1-4
https://www.elastic.co/guide/en/elasticsearch/client/javascript-api/current/api-reference-1-3.html#api-msearch-1-3

```js
client.msearch({
  body: [
    // match all query, on all indices and types
    {},
    { query: { match_all: {} } },

    // query_string query, on index/mytype
    { _index: 'myindex', _type: 'mytype' },
    { query: { query_string: { query: '"Test 1"' } } }
  ]
});
```

**Issue:** when index is defined as per reference document: '_index', elasticsearch returns query from all indexes, it does not set the index for the search.

It works if the field is set as "index": "myindex" removing the _

**Comment by gnujeremie** [comment]
same for 2.x , just went through this. I spent sometimes trying to understang why I was getting documents from other types... :-/

**Comment by spalger** [comment]
Fixed in c81168d7b9849f7d8129ad98f68aacd64ef7b5a2
